test/module/shared_model/backend_proto/proto_query_response_factory_test.cpp

```diff
@@ -0,0 +1,294 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <gtest/gtest.h>
+#include <boost/optional.hpp>
+
+#include "backend/protobuf/common_objects/proto_common_objects_factory.hpp"
+#include "backend/protobuf/proto_query_response_factory.hpp"
+#include "cryptography/crypto_provider/crypto_defaults.hpp"
+#include "framework/specified_visitor.hpp"
+#include "module/shared_model/builders/protobuf/test_block_builder.hpp"
+#include "module/shared_model/builders/protobuf/test_transaction_builder.hpp"
+#include "validators/field_validator.hpp"
+
+using namespace shared_model::proto;
+using namespace iroha::expected;
+using namespace shared_model::interface::types;
+
+using framework::SpecifiedVisitor;
+using shared_model::validation::FieldValidator;
+
+class ProtoQueryResponseFactoryTest : public ::testing::Test {
+ public:
+  std::shared_ptr<ProtoQueryResponseFactory> response_factory =
+      std::make_shared<ProtoQueryResponseFactory>();
+  std::shared_ptr<ProtoCommonObjectsFactory<FieldValidator>> objects_factory =
+      std::make_shared<ProtoCommonObjectsFactory<FieldValidator>>();
+
+  /**
+   * Put value of Result<unique_ptr<_>, _> into a shared_ptr
+   * @tparam ResultType - type of result value inside a unique_ptr
+   * @tparam ErrorType - type of result error
+   * @param res - result to be unwrapped
+   * @return shared_ptr to result value or to nullptr, if result containts error
+   */
+  template <typename ResultType, typename ErrorType>
+  std::shared_ptr<ResultType> unwrapResult(
```

Comment:
Check out `framework/result_fixture.hpp`

Reply:
It cannot be applied here, because `object_factory`, after which the method `unwrapResult` is used, returns `unique_ptr`, and `val` from framework will try to copy it